### PR TITLE
Check in parser if the source has stable IDs and if not, delete the existing cases

### PR DESCRIPTION
**Pull request by iamleeg**
Mostly putting this up as draft now as it's the easiest way to get the dockerised tests to run, which are the only way to test the function I need to change.

#1921

### CODE REVIEW

**Comment by abhidg** [comment]
Looks good, other than the default for hasStableIdentifiers which should be False as that's the case for majority. If a source actually has stable identifiers with hasStableIdentifiers, we are going to ingest the entire dataset needlessly, but it won't have incorrect data.

File: verification/curator-service/ui/package-lock.json [hunk 1]
```diff
@@ -41094,6 +40963,20 @@
         "whatwg-fetch": "^3.0.0"
       }
     },
+    "react-beautiful-dnd": {
```
Comment (abhidg) [inline verification/curator-service/ui/package-lock.json:40966]: Is this part of another PR? Seems to be from the highlighting search terms work.

Comment (iamleeg) [reply verification/curator-service/ui/package-lock.json:40966]: You know what? I don't know. I didn't explicitly put this here, I ran `npm i` to ensure the deps were up to date and I committed the change to the lock file. I can't find `react-beautiful-dnd` in the package.json so I can revert this file if you think it's wrong.

Comment (abhidg) [reply verification/curator-service/ui/package-lock.json:40966]: Right, so the highlighting work seems to be already in main, which would explain this. I'm fine with keeping this change.
```
verification/curator-service/ui/src/components/ViewCase.tsx
30:import Highlighter from 'react-highlight-words'
```

**Review by abhidg — APPROVED** [review]

**Pull request merged**